components/google-cloud/google_cloud_pipeline_components/experimental/remote/gcp_launcher/batch_prediction_job_remote_runner.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2021 The Kubeflow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""GCP launcher for batch prediction jobs based on the AI Platform SDK."""
+
+from . import job_remote_runner
+from kfp.v2.dsl import Artifact
+
+
+def create_batch_prediction_job(
+    job_type,
+    project,
+    location,
+    payload,
+    gcp_resources,
+):
+  """Create and poll batch prediction job status till it reaches a final state.
+
+  This follows the typical launching logic:
+  1. Read if the batch prediction job already exists in gcp_resources
+     - If already exists, jump to step 3 and poll the job status. This happens
+     if the launcher container experienced unexpected termination, such as
+     preemption
+  2. Deserialize the payload into the job spec and create the batch prediction
+  job
+  3. Poll the batch prediction job status every
+  job_remote_runner._POLLING_INTERVAL_IN_SECONDS seconds
+     - If the batch prediction job is succeeded, return succeeded
+     - If the batch prediction job is cancelled/paused, it's an unexpected
+     scenario so return failed
+     - If the batch prediction job is running, continue polling the status
+
+  Also retry on ConnectionError up to
+  job_remote_runner._CONNECTION_ERROR_RETRY_LIMIT times during the poll.
+  """
+  remote_runner = job_remote_runner.JobRemoteRunner(job_type, project, location,
+                                                    gcp_resources)
+
+  # Instantiate GCPResources Proto
+  job_resource = remote_runner.create_gcp_resource()
+
+  # Create batch prediction job if it does not exist
+  job_name = remote_runner.check_if_job_exists(job_resource)
+  if job_name is None:
+    job_name = remote_runner.create_job(payload, job_resource)
+
+  # Poll batch prediction job status until "JobState.JOB_STATE_SUCCEEDED"
+  remote_runner.poll_job(job_name)
+
+  return Artifact(uri="aiplatform://v1/" + job_name, name=job_name)
```

Comment:
I think this won't work as a return statement in the container code won't properly persist the artifact, in this case, to MLMD. 

The only way I found would actually work is something like this 
https://github.com/kubeflow/pipelines/blob/gcpc14/components/google-cloud/google_cloud_pipeline_components/experimental/remote/gcp_launcher/upload_model_remote_runner.py#L89

If that make sense maybe add a todo for now, and update later when that PR is merged.

Reply:
Added a TODO, will update later when that PR is merged.